### PR TITLE
get items and schema for themes

### PR DESCRIPTION
**Pull request by danbilauca**
get items and schema for themes

### CODE REVIEW

File: lib/class-wp-rest-themes-controller.php [hunk 1]
```diff
@@ -72,13 +87,32 @@ public function get_item_permissions_check( $request ) {
...
+		$slug  = $request['slug'];
+		$theme = null;
+
+		$themes = wp_get_themes();
+		foreach ( $themes as $key => $obj ) {
+			if ( $slug !== $key ) {
+				continue;
+			}
+			$theme       = $obj;
+			$theme->slug = $key;
```
Comment (marco-c) [inline lib/class-wp-rest-themes-controller.php]: I guess once we find it we can break out of the loop?

Comment (danbilauca) [inline lib/class-wp-rest-themes-controller.php]: Yep, I'll fix it in a bit

File: lib/class-wp-rest-themes-controller.php [hunk 2]
```diff
@@ -72,13 +87,32 @@ public function get_item_permissions_check( $request ) {
...
+		foreach ( $themes as $key => $obj ) {
+			if ( $slug === $key ) {
+				$theme       = $obj;
+				$theme->slug = $key;
+				break;
+			}
+		}
+
+		if ( ! $theme ) {
+			return new WP_Error( 'rest_post_invalid_id', __( 'Invalid theme slug.' ), array( 'status' => 404 ) );
```
Comment (danielbachhuber) [inline lib/class-wp-rest-themes-controller.php]: Maybe we should just use `rest_invalid` across the board.

File: lib/class-wp-rest-themes-controller.php [hunk 3]
```diff
@@ -95,12 +129,78 @@ public function delete_item( $request ) {
...
+			'tags'        => $item->get( 'Tags' ),
+		);
 	}
 
 	public function get_item_schema() {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'theme',
+			'type'       => 'object',
+			'properties' => array(
```
Comment (danielbachhuber) [inline lib/class-wp-rest-themes-controller.php]: These properties are all `readonly`, correct?

**Pull request closed without merging**